server/src/Lib.hs

```diff
@@ -80,6 +85,19 @@ data ServerConfig = ServerConfig
   }
   deriving (Show)
 
+newtype NixpkgsCommit = NixpkgsCommit { unNixpkgsCommit :: Text }
+  deriving newtype (Show)
+
+instance FromHttpApiData NixpkgsCommit where
+  parseQueryParam rawCommit = do
+    when (Text.length rawCommit  < 3) $
+      Left "too short. must be at least than 3 characters"
+    when (Text.length rawCommit  > 40) $
+      Left "too long. must be at most 40 characters"
+    unless (Text.all isHexDigit rawCommit) $
+      Left "can only consist of hexidecimal characters (0-9 and a-e)"
+    pure $ NixpkgsCommit rawCommit
```

Comment:
I think it might be useful to make a commit in other places, can you make a `String -> Either String NixpkgsCommit` smart constructor?

Reply:
Good idea, I did this in 30cabf1, although I made it based on `Text` instead of `String` since that's what `parseQueryParam` wants.